sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -159,40 +159,38 @@ object GpuHashJoin extends Arm {
    * Filter rows from the batch where any of the keys are null.
    */
   def filterNulls(cb: ColumnarBatch, boundKeys: Seq[Expression]): ColumnarBatch = {
-    var mask: ai.rapids.cudf.ColumnVector = null
+    import ai.rapids.cudf.ColumnVector
+    var mask: Option[ColumnVector] = None
     try {
       withResource(GpuProjectExec.project(cb, boundKeys)) { keys =>
         val keyColumns = GpuColumnVector.extractBases(keys)
-        keyColumns.foreach { column =>
-          if (column.hasNulls) {
-            withResource(column.isNotNull) { nn =>
-              if (mask == null) {
-                mask = nn.incRefCount()
-              } else {
-                mask = withResource(mask) { _ =>
-                  mask.and(nn)
-                }
-              }
+        mask = keyColumns
+          .filter(_.hasNulls())
+          .map(_.isNotNull())
```

Comment:
```suggestion
          .safeMap(_.isNotNull())
```
This needs to import some rapids implicits

Reply:
Why is `safemap` needed here?

Reply:
U can grep safeMap in the repo to find the cases for it.

Reply:
Add an `iterator` here, so that `map` will work in a lazy mode, and we don't need `safeMap` here. Also, this could reduce the memory usage.